### ISSUE TITLE
Enhancement: allow batching multiple operations from the command line?

### ISSUE THREAD

**Issue by felciano** [mintapi/mintapi#344]
I often find myself running `mintapi `multiple times because I'm looking grab snapshots of several datasets. For example, accounts, budgets, and transactions. Doing this via multiple command-line calls is unnecessarily slow because of the overhead of a new connection each time:

```
mintapi --budgets --filename "my_budgets.txt" <... credentials ...>
mintapi --accounts --filename "my_accounts.txt" <... credentials ...>
mintapi --extended-transactions -include-investment --filename "my_transactions.txt" <... credentials ...>
```

I've solved this with an ugly hack of cloning `api.py` and extending the code that does the command line processing, and writing a custom function that does all the steps I want within a single connection session. This is noticeably faster but ugly, hard to maintain as the mainline `api.py` evolves, and may run afoul of the various login issues keep cropping up.

The use case--batching multiple retrievals within one command-line call--seems like it is probably somewhat common, so I'd like to clean it up and contribute it, but I haven't been able to find or think up an elegant solution to it. The closest I could think of is maybe allowing multiple `config-file` parameters, each of which specifying its own operation, and having the processing code be smart about recognizing when the same Mint creds are used in each. Still feels a bit hacky tho.

Any thoughts on the merit of this enhancement idea, and if appealing, are there suggested approaches? In particular, any known / example ways to extend Python command line apps to allow grouping or batching of multiple operations within the same run of the app (and presumably using the same connection to Mint)?

**Comment by jaron-l** [comment]
Why does it need to be from the command line? Typically I would urge you to just use the python api as this package is mainly a python library. With the python api, you have control of when the session starts and ends and you make all the calls you want in between. If you writing custom python code to deal with command lines, it seems like you're already surpassed the usefulness of the command line and should just start writing a package that does what you want on top of mintapi. You could even call your python package from the command line if you really wanted to. The possible options from the command line are already large, just look at our command arguments section.

**Comment by burkematthew** [comment]
@felciano this is essentially the same request as https://github.com/mintapi/mintapi/issues/256

@jaron-l I get your point, but as we discussed on #256 , it's already done for accounts and budgets.  I think that we either need to stick with one approach or another: we need to allow simultaneous pulls of data or _only_ one.  I see both sides of the argument, especially because we don't necessarily want to assume that someone that wants to efficiently use this package is capable of writing their own python code.  If (and I do believe it can be done based on my work on #256 already) it can be implemented in a clean manner, why not support multiple data types in one request?

**Comment by felciano** [comment]
@jaron-l I have actually implemented this in python. However even my python code needs to be configured to access Mint correctly, and I've found empirically, however, is that my code is always playing catch up with `api.py` in the areas that involve configuration and connectivity. For example, it looks to me like about 50% of the command line options relate to this generic part of `mintapi`'s function -- all of these switches fall into this category:
```
      -h, --help            show this help message and exit
      --accounts            Retrieve account information (default if nothing else
                            is specified)
      --session-path [SESSION_PATH]
                            Directory to save browser session, including cookies. Used to prevent repeated
                            MFA prompts. Defaults to $HOME/.mintapi/session. Set to None to use a temporary
                            profile.
      --config-file, -c     File used to store arguments
      --filename FILENAME, -f FILENAME
                            write results to file. can be {csv,json} format.
                            default is to write to stdout.
      --keyring             Use OS keyring for storing password information
      --headless            Whether to execute chromedriver with no visible window.
      --use-chromedriver-on-path Whether to use the chromedriver on PATH, instead of downloading a local copy.
      --mfa-method {sms,email,soft-token}
                            The MFA method to automate.
      --mfa-token      The base32 encoded MFA token.
      --imap-account IMAP_ACCOUNT
      --imap-password IMAP_PASSWORD
      --imap-server IMAP_SERVER_HOSTNAME
      --imap-folder IMAP_FOLDER
                            Default is INBOX
      --imap-test           Test access to IMAP server
      --no_wait_for_sync    Do not wait for accounts to sync
      --wait_for_sync_timeout    Number of seconds to wait for sync (default is 300)
      --attention.          Get notice if there are any accounts that need attention
```
I value the ongoing work of keeping this part of `mintapi` healthy, so would like to just implement my custom functionality as an extension to these.

Also, very pragmatically, my use of `mintapi` from the command line is part of a longer script that calls other apps to process the data, many of which are not python-based. So being able to plug `mintapi` into a unix-based pipeline of processes is nice.

**Comment by felciano** [comment]
> @felciano this is essentially the same request as #256

Thanks @burkematthew I agree #256 looks very similar, and promising!

**Comment by burkematthew** [comment]
@felciano I'm actually pretty close to at least extending transactions with accounts and budgets. Hopefully once that's handled, extending even further should be no problem.

**Comment by jaron-l** [comment]
@burkematthew That's a good point. If you could publish your code, even as a draft PR, that might help me to understand.

> @jaron-l I have actually implemented this in python. However even my python code needs to be configured to access Mint correctly, and I've found empirically, however, is that my code is always playing catch up with `api.py` in the areas that involve configuration and connectivity.

@felciano, I think you misunderstand the alternate path I am suggesting. I am not suggesting that you clone mintapi and modify the code to do what you need. I'm suggesting that you write your own package that has mintapi as a dependency, makes python calls against it, and then publishes the data you need. There should be no playing catch up in this scenario as we try to not break our public api. This is also not incompatible with working in the command line as you could just call your own script from the command line and format the data however you need. An excellent package to look into would be `argparse` that could help you make a command line wrapper around mintapi. I do a similar thing in my workflow. We can try to work on batching commands together from the command line, but following this route will always allow more customization from your end while still not needing to play catch up.

**Comment by burkematthew** [comment]
@jaron-l I have a few things I'm trying to tidy up and then I can post a draft PR.

**Comment by felciano** [comment]
@burkematthew this looks very promising -- can't wait to try it out. The README says this capability is available as of "version 2.0", but the mainline still looks like it is 1.67. Is v2.0 available elsewhere or is a version bump forthcoming?

**Comment by mrooney** [comment]
@felciano we have not yet released a final 2.0, but you can always try it out from github if you want to test ahead of time :)
